### PR TITLE
modified pval input to dataframe input

### PR DESCRIPTION
**Pull request by dsrobertson**

### CODE REVIEW

File: R/ADDIS-spending.R [hunk 1]
```diff
@@ -68,20 +70,31 @@
...
+#'     'D51456', 'C88669', 'E03673', 'A63155', 'B66033'),
+#' pval = c(2.90e-08, 0.06743, 0.01514, 0.08174, 0.00171,
+#'         3.60e-05, 0.79149, 0.27201, 0.28295, 7.59e-08,
+#'         0.69274, 0.30443, 0.00136, 0.72342, 0.54757),
+#' lags = rep(1,15))
+#' 
+#' ADDIS_spending(test.df) #independent
 #' 
-#' ADDIS_spending(pval, dep=TRUE, lags=rep(1,15)) # Locally dependent
+#' ADDIS_spending(test.df, dep = TRUE) #Locally dependent
```
Comment (dsrobertson) [inline R/ADDIS-spending.R]: test.df is not defined?

File: R/ADDIS.R [hunk 2]
```diff
@@ -72,20 +73,31 @@
...
 
-ADDIS <- function(pval, alpha=0.05, gammai, w0, lambda=0.5, tau=0.5,
-                  async=FALSE, decision.times) {
+ADDIS <- function(d, alpha=0.05, gammai, w0, lambda=0.5, tau=0.5,
+                  async=FALSE) {
+    
+    if(is.data.frame(d)){
+        pval <- d$pval
+    } else if(is.vector(d)){
+        pval <- d
```
Comment (dsrobertson) [inline R/ADDIS.R:101]: Can add the following check here: if async = FALSE, then d needs to be a dataframe with a column of decision.times

File: R/LORDstar.R [hunk 3]
```diff
@@ -87,29 +90,51 @@
...
 
-LORDstar <- function(pval, alpha=0.05, version, gammai, w0,
-                decision.times, lags, batch.sizes) {
+LORDstar <- function(d, alpha=0.05, version, gammai, w0, batch.sizes) {
 
+    if(is.data.frame(d)){
+        checkSTARdf(d, version)
+        pval <- d$pval
+    } else if(is.vector(d)){
+        pval <- d
```
Comment (dsrobertson) [inline R/LORDstar.R:133]: Add the following check: if version = 'async' or 'dep', then d should be a dataframe with a column of decision.times or lags respectively

File: R/SAFFRONstar.R [hunk 4]
```diff
@@ -107,42 +110,63 @@
...
-    
-    if(version == 'async' && discard){
-        return(ADDIS(pval, alpha, gammai, w0, lambda, tau.discard,
-                 async=TRUE, decision.times))
+  
+  if(is.data.frame(d)){
+    checkSTARdf(d, version)
+    pval <- d$pval
+  } else if(is.vector(d)){
+    pval <- d
```
Comment (dsrobertson) [inline R/SAFFRONstar.R]: Add the following check: if version = 'async' or 'dep', then d should be a dataframe with a column of decision.times or lags respectively

File: R/ADDIS-spending.R (hunk 1 above)
Comment (latlio) [reply R/ADDIS-spending.R]: Done. Changed to sample.df

File: R/ADDIS.R (hunk 2 above)
Comment (latlio) [reply R/ADDIS.R:101]: Done.

File: R/LORDstar.R (hunk 3 above)
Comment (latlio) [reply R/LORDstar.R:133]: Done.

File: R/SAFFRONstar.R (hunk 4 above)
Comment (latlio) [reply R/SAFFRONstar.R]: Done

**Pull request merged**